File: src/day1/Note.java
package day1;

public class Note {

	public static void main(String[] args) {

		// Argument einlesen
		if (args.length == 0) {
			System.out
					.println("Falscher Aufruf: Bitte übergeben Sie ein Argument!");
			System.exit(-1);
		}

		int note = Integer.parseInt(args[0]);
		System.out.println("Note = " + note);

		// Wertebereich prüfen
		if ((note < 1) || (note > 6)) {
			System.out
					.println("Falscher Wert: Die Note muss einen Wert zwischen 1..6 haben!");
			System.exit(-2);
		}

		// Wert prüfen
		if (note >= 4) {
			System.out.println("Der Test ist bestanden mit:");
		} else {
			System.out.println("Der Test ist leider nicht bestanden mit:");
		}

		// Detailauswertung
		switch (note) {
		case 6:
			System.out.println("sehr gut");
			System.out.println("Herzliche Gratulation.");
			break;

		case 5:
			System.out.println("gut");
			break;

		case 4:
			System.out.println("genügend");
			break;

		default:
			System.out.println("ungenügend");
			break;
		}

	}

}

File: lab/02-grundlagen/note.py

value = input("Note = ")

try:
    note = int(value)
except Exception as e:
    print("Falsche Eingabe, Wert muss eine Zahl sein")
    print("Fehler:", e)
    exit(-1)

if note < 1 or note > 6:
    print("Falscher Wert, Zahl im Bereich 1..6 sein.")
    exit(-2)

if note >=4:
    print("Test bestanden")
else:
    print("Test nicht bestanden")

if note == 6:
    print("Note 6: sehr gut")  
elif note == 5:
    print("Note 5: gut")  
elif note == 4:
    print("Note 4: genügend")  
else:
    print("Note ungenügend")

